tests/integration2/pilot/security/authn_permissive_test.go

```diff
@@ -0,0 +1,149 @@
+	framework.Run("authn_permissive_test", m)
+}
+
+func veriyListener(listener *xdsapi.Listener, t *testing.T) bool {
+	t.Helper()
+	if listener == nil {
+		return false
+	}
+	if len(listener.ListenerFilters) == 0 {
+		return false
+	}
+	// We expect tls_inspector filter exist.
+	inspector := false
+	for _, lf := range listener.ListenerFilters {
+		if lf.Name == authnplugin.EnvoyTLSInspectorFilterName {
+			inspector = true
+			break
+		}
+	}
+	if !inspector {
+		return false
+	}
+	// Check filter chain match.
+	if len(listener.FilterChains) != 2 {
+		return false
+	}
+	mtlsChain := listener.FilterChains[0]
+	if !reflect.DeepEqual(mtlsChain.FilterChainMatch.ApplicationProtocols, []string{"istio"}) {
+		return false
+	}
+	if mtlsChain.TlsContext == nil {
+		return false
+	}
+	// Second default filter chain should have empty filter chain match and no tls context.
+	defaultChain := listener.FilterChains[1]
+	if !reflect.DeepEqual(defaultChain.FilterChainMatch, &lis.FilterChainMatch{}) {
+		return false
+	}
+	if defaultChain.TlsContext != nil {
+		return false
+	}
+	return true
+}
+
+// TestAuthnPermissive checks when authentication policy is permissive, Pilot generates expected
+// listener output.
+func TestAuthnPermissive(t *testing.T) {
+	ctx := framework.GetContext(t)
+	// TODO(incfly): make test able to run both on k8s and native when galley is ready.
+	ctx.RequireOrSkip(t, lifecycle.Test, &descriptors.NativeEnvironment, &ids.Apps)
+	env := native.GetEnvironmentOrFail(ctx, t)
+	_, err := env.ServiceManager.ConfigStore.Create(
+		model.Config{
+			ConfigMeta: model.ConfigMeta{
+				Type:      model.AuthenticationPolicy.Type,
+				Name:      "default",
+				Namespace: "istio-system",
+			},
+			Spec: &authnv1alpha.Policy{
+				// TODO: make policy work just applied to service a.
+				// Targets: []*authn.TargetSelector{
+				// 	{
+				// 		Name: "a.istio-system.svc.local",
+				// 	},
+				// },
+				Peers: []*authnv1alpha.PeerAuthenticationMethod{{
+					Params: &authnv1alpha.PeerAuthenticationMethod_Mtls{
+						Mtls: &authnv1alpha.MutualTls{
+							Mode: authnv1alpha.MutualTls_PERMISSIVE,
+						},
+					},
+				}},
+			},
+		},
+	)
+	if err != nil {
+		t.Error(err)
+	}
+	apps := components.GetApps(ctx, t)
+	a := apps.GetAppOrFail("a", t)
+	pilot := components.GetPilot(ctx, t)
+	req := appst.ConstructDiscoveryRequest(a)
+	resp, err := pilot.CallDiscovery(req)
+	if err != nil {
+		t.Errorf("failed to call discovery %v", err)
+	}
+	for _, r := range resp.Resources {
+		foo := &xdsapi.Listener{}
+		if err := proto.UnmarshalAny(&r, foo); err != nil {
+			t.Errorf("failed to unmarshal %v", err)
+		}
+		if veriyListener(foo, t) {
+			return
+		}
+	}
+	t.Errorf("failed to find any listeners having multiplexing filter chain")
+}
+
+// TestAuthentictionPermissiveE2E these cases are covered end to end
+// - app A to app B via mTLS.
+// - app A to app B via plaintext.
+// - app A to app B via https, terminating traffic at application, not sidecar.
```

Comment:
You really need to have 4 cases:
- app A to app B using plaintext (mTLS),
- app A to app B using HTTPS (mTLS),
- app A to app B using plaintext (legacy),
- app A to app B using HTTPS (legacy).

Reply:
What's the difference between `using HTTPS (mTLS),` and `using HTTPS (legacy)`?

The first is double-encrypted, HTTPS over mTLS, and the second is single HTTPS terminating at application?

Reply:
Got it. Added, thanks!